src/fides/api/oauth/system_manager.py

```diff
@@ -1,5 +1,17 @@
-from fides.common.api.scope_registry import SYSTEM_DELETE, SYSTEM_UPDATE
+from fides.common.api.scope_registry import (
+    CONNECTION_CREATE_OR_UPDATE,
+    CONNECTION_READ,
+    SYSTEM_DELETE,
+    SYSTEM_UPDATE,
+    CONNECTION_DELETE,
+)
 
 # System managers are separate from roles, because you are just granted these
 # permissions to specific system(s) not to all resources of a given type
-SYSTEM_MANAGER_SCOPES = [SYSTEM_UPDATE, SYSTEM_DELETE]
+SYSTEM_MANAGER_SCOPES = [
+    SYSTEM_UPDATE,
+    SYSTEM_DELETE,
+    CONNECTION_CREATE_OR_UPDATE,
+    CONNECTION_READ,
+    CONNECTION_DELETE,
+]
```

Comment:
Where was this happening? "a viewer can delete integrations from systems they do not own"

Reply:
Great question. For context, there are two sets of connection endpoints. The original endpoints that creation connections that aren't linked to a system and the new ones that link them.

The originals we're kept around for backwards compatibility during the migration period. Once users have completely migrated to the new flow we'll remove the original endpoints. 

The admin ui is using the new endpoints for read, create, and update but the original for deletion. Neither the original or new endpoints were validating the users role. That's how "a viewer can delete integrations from systems they do not own" was happening. 

I created a new delete route under the new `SYSTEM_CONNECTIONS_ROUTER` that validates the role and left the original unchanged since it will be removed soon.

Reply:
The ui will be updated in a follow up ticket to use the new delete route

Reply:
This isn't matching up with my understanding, maybe I'm missing something here - 

The DELETE /connection/{connection_key} endpoint has a `CONNECTION_DELETE` scope and viewers do not have the CONNECTION_DELETE scope.  - `verify_oauth_client` _does_ check roles, it just doesn't check the optional system manager permissions.  I still don't understand how a viewer could delete.

Reply:
The other one makes sense though: a viewer cannot save an integration on a system they own, we have it so you have to intentionally add a system check to the endpoints that need it.

Reply:
Thanks, for calling this out. I found out what the issue is. The BE correctly prevents a viewer from deleting a connection. The problem is the UI clears out the form regardless if the delete request is successful. I'll update the UI follow up ticket to have it fix the issue.

It only _looked_ like a viewer could delete a connection.

![image](https://github.com/ethyca/fides/assets/17103888/8ea8efea-dc03-4165-978f-76a09936f7c4)

Reply:
Ah okay! thanks for tracking this down!